File: JavaScript/hw9/Circle.js
class Circle extends Shape{
    constructor(v, r){
        super(v, r);
        this.name = 'Circle';
    }
    description(){
        var s = '{' + this.name + ';';
        s += '(' + this.o.component()[0].toString() + ',' + this.o.component()[1].toString() + ');';
        s += this.r.toString() + '}';
        return s;
    }
}

function createCircle(s) {
    s = s.split(';');
    if (s[0].charAt(0) != '(' || s[0].charAt(s[0].length - 1) != ')')
        throw new Error('Format Error')
    s[0] = s[0].substring(1, s[0].length-1);
    var v = [parseInt(s[0].split(',')[0]), parseInt(s[0].split(',')[1])];
    var r = parseInt(s[1]);
    return new Circle(new Vector(v), r);
}

File: Python/hw9/Circle.py
import Vector
import VectorInput
import Shape
import math

class Circle(Shape.Shape):
    def __init__(self, o, r):
        if r <= 0:
            raise Exception("Radius less than zero")
        self.o = o
        self.r = r

    def description(self):
        s = "{circle, "
        s += "(" + str(self.o.component(1)) + "," + str(self.o.component(2)) + ')'
        s += ', ' + str(self.r) + '}'
        return s

    def perimeter(self):
        return self.r * 2 * math.pi

    def area(self):
        return self.r * self.r * math.pi

def createCircle(so, sr):
    v = VectorInput.getVector(so)
    r = float(sr)
    if (r // 1) == r:
        r = int(r)
    o = Circle(v, r)
    return o
